Index: multicritere.php
===================================================================
RCS file: /cvsroot/ocsinventory/server/ocsreports/multicritere.php,v
retrieving revision 1.17
diff -u -r1.17 multicritere.php
--- multicritere.php	27 Feb 2008 12:34:12 -0000	1.17
+++ multicritere.php	24 Jun 2008 23:28:41 -0000
@@ -881,11 +881,13 @@
 		
 		$linSel = "LINUX"   == $_SESSION["reqs"][$label][3] ?" selected":"";
 		$winSel = "WINDOWS" == $_SESSION["reqs"][$label][3] ?" selected":"";
-		
+		$macOSX = "MacOSX"  == $_SESSION["reqs"][$label][3] ?" selected":"";
+
 		if( $champ=="osname")
 			echo "<option value='LINUX' $linSel>LINUX (".$l->g(547).")</option>
-				  <option value='WINDOWS' $winSel>WINDOWS (".$l->g(547).")</option>";
-		
+				  <option value='WINDOWS' $winSel>WINDOWS (".$l->g(547).")</option>
+				  <option value='MacOSX' $macOSX>MacOSX (".$l->g(547).")</option>";
+
 		while($row=mysql_fetch_array($res))
 		{
 			if($row[0]=="") continue;	
